### DISCUSSION TITLE
DORA Core

### DISCUSSION THREAD

**Discussion by davidstanke** [dora-team/dora.dev#265] [Ideas]
We're introducing DORA Core, a simplified presentation of DORA's research program, for easier application by practitioners ([project plan](https://github.com/orgs/dora-team/projects/3)). Use this space to discuss DORA Core as it matures.

(also see the FAQ at [dora.dev/core](https://dora.dev/core))

**Comment by davidstanke** [comment]
Launch plan:

- [x] publish [dora.dev/core](https://dora.dev/core)
- [x] add modal pop-ups with short descriptions to each item on the Model
- [x] add links from modals to full capability pages
- [x] add capability pages for core entities that don't yet have them
- #341 
- #342

**Comment by punkstarman** [comment]
It's a shame that Lean capabilities aren't included. Will they be added in the future?

**Reply by davidstanke** [reply]
Agreed! Lean capabilities showed great promise but we haven't been able (yet) to return to and reinforce that line of inquiry, so we don't have sufficient evidence to include them in Core. 

Lean didn't make the cut for 2023 research, but I'm glad to hear there's interest—maybe we can return to it in next year's study.

**Comment by tonit** [comment]
Thank you for working on this matter! I guess this is not just @davidstanke work but thanks thanks thanks! Great to see the whole DevEx space accelerate (pun intended).

Seeing the Capabilities exaplained too, this is my (new) go-to resource showing with clients at the moment. Huge thanks!

Let me do some remarks, not in a particular order:
- I am missing motivational capabilities a lot. (intrinsic and extrinsic) Not sure if this is covered by the generative organizational culture (big one) or not. I feel motivation is both a capability and an outcome (forming a loop).
- Will you share the diagramm in a text/diffable manner too? Or maybe the (drawio?) source? In order to contribute on this, too.. I see that you have it versioned already (version 0.2). Writing that i see it is Adobe Illustrator. As it probably will grow, what are the chances to make it more "open"?

**Reply by davidstanke** [reply]
@tonit I'm glad to hear that you're already making use of it! And you raise some really good points; here are my thoughts...

* _motivation:_ Yes! In our consultative work with teams, we talk a lot about motivations and incentives, but you're right that these concepts aren't deeply reflected in the Core model. Generative culture is adjacent, but I'd say a more relevant topic is one that hasn't [yet] graduated to Core: [Transformational Leadership](https://dora.dev/devops-capabilities/cultural/transformational-leadership/). Going forward, this is certainly a ripe area for investigation. (You also might find [this article on productivity](https://queue.acm.org/detail.cfm?id=3595878) interesting.)
* _diagram source:_ Yes and yes... [the source](https://github.com/dora-team/dora.dev/blob/main/resources/graphics/dora-core-model.ai) of the current/first version is actually not even _really_ Adobe Illustrator; it's a PDF export from Google Slides that I passed through Illustrator for some minor tweaks. Going forward, I absolutely want to use a more open/textual source format, and anything that can be versioned into the repo will be. Drawio may be an option, though I haven't investigated it yet. 
  * The ideal format would be something that starts as machine-parseable text, which is then exported into a format (svg?) that can be decorated via CSS and programmatically enhanced with Javascript to support things like modal popups for each entity.
  * The current ["spider diagram"](https://dora.dev/research) uses [D3](https://d3js.org/), which checks a lot of these boxes, but I don't like how the rendered visualization is non-deterministic: the arrangement of nodes in the graph changes based on the viewport and user interaction. Per the intent of Core, I would like to have a standard, consistent visualization of the model across users and clients. However, in fairness to D3, I haven't yet researched if/how it might support such determinism.

**Reply by davidstanke** [reply]
Update -- as of now, I'm still using a design-oriented tool (it's now Figma) to create the SVG. Having a consistent, easy-to-read layout is a priority, and using Figma is a shortcut to that. However, for future versions, I'm very much open to the idea of using something machine-generated based on a parseable text format. But it's gotta be at least as pretty as it is now. :)

FYI, the current state can be seen at dora.dev/research/research-with-core

**Comment by davidstanke** [comment]
UPDATE: Core is now live! Find it at dora.dev/research -- a big thank you to everyone in the extended community who has helped to create this first iteration.

Of course we will always be improving it, so I will leave this discussion open.

* The concepts and connections will be revisited ~annually, starting in Q1 2024
* The visualization/interactive model will be continuously revised. Found a bug? [Report it!](https://github.com/dora-team/dora.dev/issues) Got ideas or questions? Post them in this discussion.

**Comment by Winael** [comment]
@davidstanke Talking with lots of practionners as Coach DevOps, and we are all agree that Transformational Leadership is really mandatory to be able to achieve a DevOps transformation. And someway, it's something you're talkink about in the new article from DORA: we need leaders sponsor but also a management that help to put in place technical and lean capabilities. There's lots of research in management science that already proove the benefits of Transformational Leadership.

I suppose it will be add in a next release :)

Anyway, thanks again for all you works, it's amazing :)

**Comment by davidstanke** [comment]
Open question: should we change the language of the "TTR" metric to "Failed Deployment Recovery Time"? See https://github.com/dora-team/dora.dev/issues/449

**Reply by steve-fenton-octopus** [reply]
+1 for making this consistently "Failed Deployment Recovery Time"

**Comment by davidstanke** [comment]
It's time for the next version of Core! This discussion is now closed and we'll continue in [DORA Core V2 Discussion](https://github.com/dora-team/dora.dev/discussions/582)
